openedx/openedx-platform
# SE-1017 Optionally make registration fields read-only

This PR adds the ability to optionally disable/make uneditable the registration form fields when using third party authentication. This feature can be enabled for any provider by adding `PROVIDER_READ_ONLY_FIELDS` options under `Other Settings` (incase of OAuth2) or `Advanced Settings` (incase of SAML), which contains the list of fields to be made read-only.

**Dependencies:** None

**Sandbox URL:** 

- LMS:  https://pr20957.sandbox.opencraft.hosting/

- Studio: https://studio.pr20957.sandbox.opencraft.hosting/

**Testing Instructions:**
- Set up third party auth/SSO on the sandbox. Ensure to enable the `send to registration form` flag in the provider config using the django admin.
- Add the registration field names that you'd like to make read-only under the advanced settings like:
```
{"PROVIDER_READ_ONLY_FIELDS": ["email", "name"]}
```
- Register a new account using the SSO.
- Verify that the fields made read_only above, is not editable.

**Author notes and concerns:**
Currently it is assumed that only the default registration fields (i.e. name, email and password) are used. Incase additional fields are used, the [restrictions](https://github.com/open-craft/edx-platform/blob/8457746a05f10b107a247b6f09873b58db8c48b9/openedx/core/djangoapps/user_api/helpers.py#L126-L133) should be updated accordingly.
